_dprhtml/js/search.test.js

```diff
@@ -0,0 +1,39 @@
+import DPR_G from "./globalObject";
+import { searchTipitaka, SearchType } from "./search";
+
+describe("searchTipitaka", () => {
+  beforeEach(() => {
+    $('<div/>', { id: 'finished' }).appendTo('body');
+    $('<div/>', { id: 'sbfab' }).appendTo('body');
+    $('<div/>', { id: 'sbfb' }).appendTo('body');
+    $('<div/>', { id: 'sbfbc' }).appendTo('body');
+    $('<div/>', { id: 'searchb' }).appendTo('body');
+    $('<div/>', { id: 'search-link' }).appendTo('body');
+    $('<div/>', { id: 'search-sets' }).appendTo('body');
+  });
+
+  test("generates the correct permalink when searching", async () => {
```

Comment:
Despite being a pretty heavyweight integration test, this test runs fairly quickly (~80ms on my machine). ⚡

Reply:
Doesn't sanity check the page output but that can wait for the next bug.

Reply:
@parthopdas I added some extra expectations to make sure the search results are correct. See https://github.com/digitalpalireader/digitalpalireader/pull/345/files#diff-7d6aa5bc48dce40ad0fbc3e8127451146239510fbb30733d00016650561f5125R38-R43.